ui/src/app/(index)/page.tsx

```diff
@@ -1,6 +1,69 @@
-const Home = () => {
+import Image from 'next/image';
+import BeforeLogin from '@/app/(index)/_components/BeforeLogin';
+import AfterLogin from '@/app/(index)/_components/AfterLogin';
+import { getCurrentUser } from '@/access/AuthenticationService';
+import Role from '@/access/Role';
+import LoginButton from '@/app/(index)/_components/LoginButton';
+import {getAnnouncements, getNumberOfGroupings, getNumberOfMemberships} from '@/services/GroupingsApiService';
+import Announcement from '@/app/(index)/_components/Announcement';
+
+const Home = async () => {
+    const [currentUser, numberOfGroupings, numberOfMemberships, announcements] = await Promise.all([
+        getCurrentUser(),
+        getNumberOfGroupings(),
+        getNumberOfMemberships(),
+        getAnnouncements()
+    ]);
+
+    const activeAnnouncements = announcements?.announcements
+        ?.filter((announcement) => announcement.state === 'Active')
+        ?.map((announcement) => announcement.message) || [];
```

Comment:
There should be a way around using the `?`. Usually you can put an if statement to check if announcements is not null/undefined/empty. Then the `?` would not be needed.